### PR TITLE
Use useRouter over Router for with-google-analytics example

### PR DESCRIPTION
**Pull request by chibicode**
Updating `with-google-analytics` example to be in line with [our documentation recommending `useRouter`](https://nextjs.org/docs/api-reference/next/router#userouter). Verified that it works.

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| buildDuration | 12.5s | 12.7s | ⚠️ +206ms |
| nodeModulesSize | 56.7 MB | 56.7 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.464 | 2.386 | -0.08 |
| / avg req/sec | 1014.6 | 1047.78 | +33.18 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.426 | 1.373 | -0.05 |
| /error-in-render avg req/sec | 1753.38 | 1821.26 | +67.88 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| 677f882d2ed8..b5de.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-ec49a3f..d437.js gzip | 7.08 kB | 7.08 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.4 kB | 6.4 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-b3aecf0..dule.js gzip | 6.14 kB | 6.14 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-d2344ce..8b36.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-f8c0daf..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| index.html gzip | 971 B | 971 B | ✓ |
| link.html gzip | 976 B | 976 B | ✓ |
| withRouter.html gzip | 963 B | 963 B | ✓ |
| Overall change | 2.91 kB | 2.91 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| buildDuration | 14.1s | 14.5s | ⚠️ +374ms |
| nodeModulesSize | 56.7 MB | 56.7 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| 677f882d2ed8..b5de.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-ec49a3f..d437.js gzip | 7.08 kB | 7.08 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.4 kB | 6.4 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-b3aecf0..dule.js gzip | 6.14 kB | 6.14 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-d2344ce..8b36.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-f8c0daf..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _error.js | 1.03 MB | 1.03 MB | ✓ |
| 404.html | 4.22 kB | 4.22 kB | ✓ |
| hooks.html | 3.86 kB | 3.86 kB | ✓ |
| index.js | 1.03 MB | 1.03 MB | ✓ |
| link.js | 1.08 MB | 1.08 MB | ✓ |
| routerDirect.js | 1.07 MB | 1.07 MB | ✓ |
| withRouter.js | 1.07 MB | 1.07 MB | ✓ |
| Overall change | 5.29 MB | 5.29 MB | ✓ |

</details>

</details>
Commit: 955e6f3cbea8fbd344e0cc5dbda6f187dd3e4658

File: examples/with-google-analytics/pages/_app.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
       gtag.pageview(url)
     }
-    Router.events.on('routeChangeComplete', handleRouteChange)
+    router.events.on('routeChangeComplete', handleRouteChange)
     return () => {
-      Router.events.off('routeChangeComplete', handleRouteChange)
+      router.events.off('routeChangeComplete', handleRouteChange)
     }
-  }, [])
+  }, [router.events])
```
Comment (chibicode) [inline examples/with-google-analytics/pages/_app.js:15]: `router.events` was added with `exhaustive-deps`. I think this is fine?

Comment (lfades) [reply examples/with-google-analytics/pages/_app.js:15]: Yes

**Review by lfades — APPROVED** [review]
Thank you!

**Review by ijjk — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| buildDuration | 13.7s | 14s | ⚠️ +283ms |
| nodeModulesSize | 56.7 MB | 56.7 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.605 | 2.585 | -0.02 |
| / avg req/sec | 959.57 | 967.28 | +7.71 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.532 | 1.606 | ⚠️ +0.07 |
| /error-in-render avg req/sec | 1631.38 | 1556.45 | ⚠️ -74.93 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| 677f882d2ed8..b5de.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-195c642..b6b7.js gzip | 7.08 kB | 7.08 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.4 kB | 6.4 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ad445e5..dule.js gzip | 6.14 kB | 6.14 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-d2344ce..8b36.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-f8c0daf..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| index.html gzip | 968 B | 968 B | ✓ |
| link.html gzip | 975 B | 975 B | ✓ |
| withRouter.html gzip | 962 B | 962 B | ✓ |
| Overall change | 2.9 kB | 2.9 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| buildDuration | 16s | 15.4s | -607ms |
| nodeModulesSize | 56.7 MB | 56.7 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| 677f882d2ed8..b5de.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-195c642..b6b7.js gzip | 7.08 kB | 7.08 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.4 kB | 6.4 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-ad445e5..dule.js gzip | 6.14 kB | 6.14 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-ed1b0..8fbd.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-d2344ce..8b36.js gzip | 1.3 kB | 1.3 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-f8c0daf..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 329 B | 329 B | ✓ |
| Overall change | 651 B | 651 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | chibicode/next.js hotfix-router-with-google-analytics | Change |
| - | - | - | - |
| _error.js | 1.03 MB | 1.03 MB | ✓ |
| 404.html | 4.22 kB | 4.22 kB | ✓ |
| hooks.html | 3.86 kB | 3.86 kB | ✓ |
| index.js | 1.03 MB | 1.03 MB | ✓ |
| link.js | 1.08 MB | 1.08 MB | ✓ |
| routerDirect.js | 1.07 MB | 1.07 MB | ✓ |
| withRouter.js | 1.07 MB | 1.07 MB | ✓ |
| Overall change | 5.29 MB | 5.29 MB | ✓ |

</details>

</details>
Commit: b6b1ed605d7507fc11146befe49bb06458b2b4ad

**Comment by balazsorban44** [comment]
@chibicode what's the gain here?

**Comment by lfades** [comment]
@balazsorban44 We no longer document or recommend the usage of `useRouter` over importing `Router`.

**Comment by balazsorban44** [comment]
@lfades Yeah, I understand that, but is there a reason, like maybe removing the Router import possibility I in the future? Or any performance gains? Or is it simply not the "React way"?

Just curious/trying to understand. :)

**Comment by lfades** [comment]
@balazsorban44 With `useRouter` you have to inside of a component in order to use the router and we would like to keep it that way, using the router outside of a component will lead you to unexpected results.

**Pull request merged**